EventFilter/GEMRawToDigi/interface/AMCdata.h

```diff
@@ -0,0 +1,259 @@
+	(static_cast<uint64_t>(m_Rtype & 0x0f) <<  56) |
+	(static_cast<uint64_t>(m_Param1) << 48) |
+	(static_cast<uint64_t>(m_Param2) << 40) |
+	(static_cast<uint64_t>(m_Param3) << 32) |
+	(static_cast<uint64_t>(m_Onum) << 16) |
+	(static_cast<uint64_t>(m_BID));
+    }
+    
+    //!Reads the word for the GEM Event Header
+    // Fills the GEM DAV list, Buffer Status, GEM DAV count, and TTS state.
+    void setGEMeventHeader(uint64_t word)
+    {
+      m_GEMDAV = 0x00ffffff & (word >> 40);   /*!<GEM DAV list*/
+      m_Bstatus = 0x00ffffff & (word >> 16);  /*!<Buffer Status*/
+      m_GDcount = 0b00011111 & (word >> 11);  /*!<GEM DAV count*/
+      m_Tstate = 0b00000111 & word;           /*!<TTS state*/
+    }
+    uint64_t getGEMeventHeader() const
+    {
+      return
+	(static_cast<uint64_t>(m_GEMDAV & 0x00ffffff) << 40) |
+	(static_cast<uint64_t>(m_Bstatus & 0x00ffffff) << 16) |
+	(static_cast<uint64_t>(m_GDcount & 0b00011111) << 11) |
+	(static_cast<uint64_t>(m_Tstate & 0b00000111));
+    }
+
+    //!Reads the word for the GEM Event Trailer
+    // Fills the Chamber Timeout and OOS GLIB.
+    void setGEMeventTrailer(uint64_t word)
+    {
+      m_ChamT = 0x00ffffff & (word >> 40);  /*!<Chamber Timeout*/
+      m_OOSG = 0b00000001 & (word >> 39);   /*!<OOS GLIB*/
+    }
+    uint64_t getGEMeventTrailer() const
+    {
+      return
+	(static_cast<uint64_t>(m_ChamT & 0x00ffffff) << 40) |
+	(static_cast<uint64_t>(m_OOSG & 0b00000001) << 39);
+    }
+
+    //!Reads the word for the AMC Trailer
+    void setAMCTrailer(uint64_t word)
+    {
+      m_CRC = word >> 32;
+      m_L1AT = word >> 24;
+      m_DlengthT = 0x000fffff & word;
+    }
+    uint64_t getAMCTrailer() const
+    {
+      return
+	(static_cast<uint64_t>(m_CRC) << 32) |
+	(static_cast<uint64_t>(m_L1AT) << 24) |
+	(static_cast<uint64_t>(m_DlengthT & 0x000fffff));
+    }
+
+    uint8_t  amcNum()   const {return m_AMCnum;}   ///<Returns AMC number
+    uint32_t l1A()      const {return m_L1A;}      ///<Returns L1A number
+    uint16_t bx()       const {return m_BX;}       ///<Returns Bunch Crossing ID
+    uint32_t dlength()  const {return m_Dlength;}  ///<Returns Data Length (Overall size of FED event fragment)
+		         
+    uint8_t  fv()       const {return m_FV;}       ///<Returns Format Version
+    uint8_t  runType()  const {return m_Rtype;}    ///<Returns Run Type
+    uint8_t  param1()   const {return m_Param1;}   
+    uint8_t  param2()   const {return m_Param2;}
+    uint8_t  param3()   const {return m_Param3;}
+    uint16_t orbitNum() const {return m_Onum;}     ///<Returns Orbit number
+    uint16_t boardId() const {return m_BID;}      ///<Returns Board ID
+
+    uint32_t gemDAV ()  const {return m_GEMDAV;}        ///<Returns GEM DAV list (which chambers have data)
+    uint64_t bStatus()  const {return m_Bstatus;}       ///<Returns Buffer status
+    int      gdCount()  const {return unsigned(m_GDcount);} ///<Returns GEM DAV count (number of chamber blocks)
```

Comment:
Why cast an uint8_t to unsigned and return it as int?

Reply:
@mommsen - changed to unique_ptrs